cdap-data-fabric/src/main/java/co/cask/cdap/data/runtime/DataSetServiceModules.java

```diff
@@ -171,4 +173,30 @@ protected void configure() {
       }
     };
   }
+
+  private static final class DatasetMdsProvider implements Provider<DatasetFramework> {
```

Comment:
can we NOT use provider? Seems like we can do bindings without it, no? Making guice module complex doesn't do good